src/java/org/apache/cassandra/io/util/MmappedRegions.java

```diff
@@ -140,6 +147,25 @@ public void extend(long length)
         copy = new State(state);
     }
 
+    public void extend(CompressionMetadata compressionMetadata)
+    {
+        assert !isCopy() : "Copies cannot be extended";
+
+        if (compressionMetadata.dataLength <= state.length)
```

Comment:
Nit: there should be a comment why `dataLength` is used here and `compressedFileLength` below. (If I'm not mistaken the data length might be truncated to shorter than the end of the last chunk, while we do want to map the whole chunk to be able to read it).

Reply:
I'll add a comment

Reply:
Actually, there is no comment to add. It is simply a bug. Not a harmful one, but a bug.

Reply:
https://github.com/apache/cassandra/pull/2064/commits/5022c14d3583666f936c3467b4594bade7fbd360